### PR TITLE
Add loading and error pages

### PR DESCRIPTION
**Pull request by ghickman**
This adds a loading and error page for the Electron side of the app. It also unifies how we load the server URL along the way to make it easier to set what the app should do on success/failure.

I tried to use node's [`util.promisify()`](https://nodejs.org/api/util.html#utilpromisifyoriginal) but it threw up a native modal with the error rather than invoking my code in `.catch()` and I couldn't work out how I was holding it wrong.

I've copied the styles Tom added in #217.

Ref: #12

### CODE REVIEW

File: sacro-app/src/create-window.js [hunk 1]
```diff
@@ -9,9 +9,30 @@ const { mainMenu } = require("./main-menu");
...
+  }
+
+  return win;
+}
+
 let TEMPDIR = null;
 
 const createWindow = async () => {
+  const error = windowBuilder({ file: "error.html", show: false });
+  const splash = windowBuilder({
```
Comment (bloodearnest) [inline sacro-app/src/create-window.js:30]: Huh, this looks very different to what I'd expected. We have 3 separate windows?

I'd naively assumed we would:

1) create the window
2) `win.loadUrl("file:///path/to/splash.html")`
3) wait for server
 - on success `win.loadUrl(serverUrl)`
 - on error `win.loadUrl("file:///path/to/error.html")`

Does that not work?

Comment (ghickman) [reply sacro-app/src/create-window.js:30]: Hah!  Hadn't even occurred to me to test changing the loaded resource for the current window.  That works and simplifies some things 👍

Comment (bloodearnest) [reply sacro-app/src/create-window.js:30]: It's the much vaunted programmable hypermedia client! :magic_wand:

**Review by bloodearnest — APPROVED** [review]
Much nicer :+1:

**Pull request merged**